mgplvm/kernels/stationary.py

```diff
@@ -0,0 +1,309 @@
+import abc
+import math
+import torch
+from torch import nn, Tensor
+from ..utils import softplus, inv_softplus
+from .kernel import Kernel
+from typing import Tuple, List
+import numpy as np
+
+
+class Stationary(Kernel, metaclass=abc.ABCMeta):
+
+    def __init__(self,
+                 n: int,
+                 distance,
+                 d=None,
+                 ell=None,
+                 scale=None,
+                 learn_scale=True,
+                 Y: np.ndarray = None):
+        """
+        Parameters
+        ----------
+        n : int 
+            number of batches (neurons)
+        distance : 
+            distance function
+        d : Optional[int]
+            dimension of the input variables
+            if provided, there is a separate length scale for each input dimension
+        ell: Optional[np.ndarray]
+            lengthscale hyperparameter
+            it should have dimensions n x d if d is not None and n if d is None
+        scale : Optional[np.ndarray]
+            scale hyperparameter
+            it should have dimension n 
+        learn_scale : bool
+            optimises the scale hyperparameter if true
+        Y : Optional[np.ndarray]
+            data matrix used for initializing the scale hyperparameter
+        """
+
+        super(Stationary, self).__init__()
+
+        if scale is not None:
+            _scale = torch.tensor(scale, dtype=torch.get_default_dtype()).sqrt()
+        elif Y is not None:
+            _scale = torch.tensor(np.mean(Y**2, axis=(0, -1))).sqrt()
+        else:
+            _scale = torch.ones(n,)
+
+        self._scale = nn.Parameter(data=_scale, requires_grad=learn_scale)
+
+        self.ard = d is not None
+        if ell is None:
+            if d is None:
+                _ell = inv_softplus(torch.ones(n,))
+            else:
+                assert (d is not None)
+                _ell = inv_softplus(torch.ones(n, d))
```

Comment:
I think it would be good to provide the option of having ell.shape = (d); i.e. learning a separate length scale per latent dimension but not per neuron. This should give a clearer interpretation of the latent dimensionality. Thoughts?

Reply:
I think that's a great idea, maybe for the next PR?

Reply:
sure thing